src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer {

```

Comment:
The `MachineProducer` interface should be parameterized to ensure type safety. According to the task requirements, you should replace `Object` in `get()` with the suitable option, such as `MachineProducer<Truck>`.